plugins/kubernetes/app/models/kubernetes/release.rb

```diff
@@ -74,6 +69,21 @@ def create_release_docs(params)
       raise 'No Kubernetes::ReleaseDoc has been created' if release_docs.empty?
     end
 
+    def release_doc_for(deploy_group_id, role_id)
+      release_docs.find { |doc| doc.kubernetes_role.id == role_id && doc.deploy_group.id == deploy_group_id }
+    end
+
+    def update_status(release_doc)
+      case
+      when release_docs.all?(&:live?) then self.status = :live
+      when release_docs.all?(&:dead?) then self.status = :dead
+      when release_doc.spinning_up? then self.status = :spinning_up
+      when release_doc.spinning_down? then self.status = :spinning_down
+      when release_docs.any?(&:dead?) then self.status = :spinning_down
+      end
+      save!
+    end
```

Comment:
I think we shouldn't. From what I understood, ReleaseDocs gets updated to failed when we get a failed Pod. Releases get updated to fail when we reach the timeout. Am I correct? If so, I think we should ignore a `failed` ReleaseDoc and not update the Release.

Were you doing something in here? Did I miss something in the merge process?